modules/gateway/outputs.tf

```diff
@@ -24,10 +24,32 @@ output "public_ip_address_fqdn" {
 
 output "gateway_id" {
   description = "Identifier of the application gateway for GraphDB"
-  value       = var.gateway_enable_private_access ? azurerm_application_gateway.graphdb-private[0].id : azurerm_application_gateway.graphdb-public[0].id
+  value       = var.gateway_enable_private_access && length(azurerm_application_gateway.graphdb-private) > 0 ? azurerm_application_gateway.graphdb-private[0].id : !var.gateway_enable_private_access && length(azurerm_application_gateway.graphdb-public) > 0 ? azurerm_application_gateway.graphdb-public[0].id : null
 }
 
+# Gateway Backend Adress Pool ID:
+# Check if application gateway is enabled and private access is true. If both are available, checks for backend adress pool and retrieves the ID of the first one. Otherwise returns Null.
+# If private access is not enabled, if a public gateway exists with a backend address pool and retrieves its ID if present.
```

Comment:
If Private access is disabled and there is a Public Gateway, retrieves the first backend address pool ID.

Reply:
changed